### ISSUE TITLE
Allow self signed certificates to be used for temporarily spawned registries

### ISSUE THREAD

**Issue by masinger** [masinger/testcontainers-java#2]
If testcontainers are run to often, it is quite common to exceed the rate limits of LetsEncrypt. In order to circumvent this issue by providing a custom (self sigend) certificate, it is necessary to allow for "insecure" Kaniko-Builds.

```
Caused by:
        java.lang.IllegalStateException: Failed to load ApplicationContext
            at org.springframework.test.context.cache.DefaultCacheAwareContextLoaderDelegate.loadContext(DefaultCacheAwareContextLoaderDelegate.java:132)
            at org.springframework.test.context.support.DefaultTestContext.getApplicationContext(DefaultTestContext.java:123)
            at org.springframework.test.context.junit.jupiter.SpringExtension.getApplicationContext(SpringExtension.java:207)
            at org.springframework.test.context.junit.jupiter.SpringExtension.resolveParameter(SpringExtension.java:193)
            at org.junit.jupiter.engine.execution.ExecutableInvoker.resolveParameter(ExecutableInvoker.java:216)
            ... 75 more

            Caused by:
            io.fabric8.kubernetes.client.KubernetesClientException: Failure executing: POST at: https://10.43.0.1/apis/apps/v1/namespaces/testcontainers-foo-lboin/deployments. Message: deployments.apps "repository" already exists. Received status: Status(apiVersion=v1, code=409, details=StatusDetails(causes=[], group=apps, kind=deployments, name=repository, retryAfterSeconds=null, uid=null, additionalProperties={}), kind=Status, message=deployments.apps "repository" already exists, metadata=ListMeta(_continue=null, remainingItemCount=null, resourceVersion=null, selfLink=null, additionalProperties={}), reason=AlreadyExists, status=Failure, additionalProperties={}).
                at io.fabric8.kubernetes.client.dsl.base.OperationSupport.requestFailure(OperationSupport.java:686)
                at io.fabric8.kubernetes.client.dsl.base.OperationSupport.assertResponseCode(OperationSupport.java:625)
                at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:565)
                at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:526)
                at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleCreate(OperationSupport.java:314)
                at io.fabric8.kubernetes.client.dsl.base.BaseOperation.handleCreate(BaseOperation.java:753)
                at io.fabric8.kubernetes.client.dsl.base.BaseOperation.handleCreate(BaseOperation.java:88)
                at io.fabric8.kubernetes.client.dsl.base.CreateOnlyResourceOperation.create(CreateOnlyResourceOperation.java:61)
                at org.testcontainers.providers.kubernetes.repository.TemporaryImageRepositoryStrategy.setup(TemporaryImageRepositoryStrategy.java:110)
                at org.testcontainers.providers.kubernetes.repository.TemporaryImageRepositoryStrategy.getRegistryIngress(TemporaryImageRepositoryStrategy.java:65)
                at org.testcontainers.providers.kubernetes.repository.TemporaryImageRepositoryStrategy.getBaseImagePrefix(TemporaryImageRepositoryStrategy.java:59)
                at org.testcontainers.providers.kubernetes.repository.TemporaryImageRepositoryStrategy.getRandomImageName(TemporaryImageRepositoryStrategy.java:55)
                at org.testcontainers.providers.kubernetes.KubernetesContainerController.getRandomImageName(KubernetesContainerController.java:72)
                at org.testcontainers.images.builder.ImageFromDockerfile.<init>(ImageFromDockerfile.java:57)
                at com.foo.TestContainerInitializer.initialize(TestContainerInitializer.kt:36)
```

**Comment by masinger** [comment]
# Update: Not a race condition
The problem is not a race condition, but a failing ACME challenge (caused by hitting the max certificates per domain rate limit)

´´´
 java.lang.RuntimeException: Timed out waiting for TLS to be available
    	at org.testcontainers.providers.kubernetes.repository.TemporaryImageRepositoryStrategy.setup(TemporaryImageRepositoryStrategy.java:199) ~[providers-kubernetes-kubernetes-alpha0.5.jar:na]
    	at org.testcontainers.providers.kubernetes.repository.TemporaryImageRepositoryStrategy.getRegistryIngress(TemporaryImageRepositoryStrategy.java:65) ~[providers-kubernetes-kubernetes-alpha0.5.jar:na]
    	at org.testcontainers.providers.kubernetes.repository.TemporaryImageRepositoryStrategy.getBaseImagePrefix(TemporaryImageRepositoryStrategy.java:59) ~[providers-kubernetes-kubernetes-alpha0.5.jar:na]
    	at org.testcontainers.providers.kubernetes.repository.TemporaryImageRepositoryStrategy.getRandomImageName(TemporaryImageRepositoryStrategy.java:55) ~[providers-kubernetes-kubernetes-alpha0.5.jar:na]
```
```
cert-manager/controller/orders "msg"="failed to create Order resource due to bad request, marking Order as failed" "error"="429 urn:ietf:params:acme:error:rateLimited: Error creating new order :: too many certificates already issued for: ******.de: see https://letsencrypt.org/docs/rate-limits/" "resource_kind"="Order" "resource_name"="registry-cert-mbwsy-wbj7f-2128556337" "resource_namespace"="testcontainers-testo-mbwsy" "resource_version"="v1"
```

**Comment by masinger** [comment]
Resolved by #1
